### PR TITLE
[BugFix] fix delta column read when using bitmap & remove pk column from delta column

### PR DESCRIPTION
**Pull request by luohaha**
## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #20436

## Problem Summary(Required):
This PR contains two parts:
1. It fix the bug that bitmap on delta column segment not work.
2. It remove the pk column from delta column group when update, to save disk space.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by decster — DISMISSED** [review]

File: be/src/storage/rowset/segment_iterator.cpp [hunk 1]
```diff
@@ -420,23 +424,36 @@ Status SegmentIterator::_try_to_update_ranges_by_runtime_filter() {
...
+    }
+    // the column not exist in delta column group
+    return nullptr;
+}
+
+StatusOr<std::unique_ptr<ColumnIterator>> SegmentIterator::_new_dcg_column_iterator(uint32_t ucid,
+                                                                                    std::string* filename) {
+    // build column iter from delta column group
+    int32_t col_index = 0;
+    ASSIGN_OR_RETURN(auto dcg_segment, _get_dcg_segment(ucid, &col_index));
```
Comment (chaoyli) [inline be/src/storage/rowset/segment_iterator.cpp:451]: Does the bitmap index exist in the main segment, or in the delta column group?

File: be/src/storage/rowset_column_update_state.cpp [hunk 2]
```diff
@@ -422,19 +429,24 @@ Status RowsetColumnUpdateState::finalize(Tablet* tablet, Rowset* rowset, const P
     RETURN_IF_ERROR(tablet->updates()->get_latest_applied_version(&latest_applied_version));
     RETURN_IF_ERROR(_finalize_partial_update_state(tablet, rowset, latest_applied_version, index));
 
-    std::vector<uint32_t> update_column_ids;
-    std::vector<int32_t> update_column_indexes;
+    std::vector<int32_t> update_column_ids;
+    std::vector<uint32_t> update_column_ids_u;
```
Comment (chaoyli) [inline be/src/storage/rowset_column_update_state.cpp:433]: `update_column_uids` is a better name.

File: be/src/storage/rowset_column_update_state.cpp [hunk 3]
```diff
@@ -96,12 +96,17 @@ Status RowsetColumnUpdateState::_load_upserts(Rowset* rowset, uint32_t idx) {
         pk_columns.push_back((uint32_t)i);
     }
     std::vector<uint32_t> update_columns;
+    std::vector<uint32_t> update_columns_with_keys;
     const auto& txn_meta = rowset->rowset_meta()->get_meta_pb().txn_meta();
     for (uint32_t cid : txn_meta.partial_update_column_ids()) {
-        update_columns.push_back(cid);
+        update_columns_with_keys.push_back(cid);
```
Comment (chaoyli) [inline be/src/storage/rowset_column_update_state.cpp:102]: If remove the keys from the delta group, how to corresponds the delta group with main segment?

File: be/src/storage/rowset/segment_iterator.cpp (hunk 1 above)
Comment (luohaha) [reply be/src/storage/rowset/segment_iterator.cpp:451]: The old columns's bitmap index exist in the main segment, and the new columns(after update)'s bitmap index exist in the delta column group.

File: be/src/storage/rowset_column_update_state.cpp (hunk 3 above)
Comment (luohaha) [reply be/src/storage/rowset_column_update_state.cpp:102]: The rows in delta column is already sorted, and correspond with rows in main segment one by one.

File: be/src/storage/rowset_column_update_state.cpp (hunk 2 above)
Comment (luohaha) [reply be/src/storage/rowset_column_update_state.cpp]: Done

**Review by chaoyli — APPROVED** [review]

**Review by decster — APPROVED** [review]

**Pull request merged**